server/src/test/java/org/eclipse/lsp/cobol/domain/databus/impl/DefaultDataBusBrokerTest.java

```diff
@@ -0,0 +1,147 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.domain.databus.impl;
+
+import com.google.common.eventbus.DeadEvent;
+import com.google.common.eventbus.Subscribe;
+import org.eclipse.lsp.cobol.domain.databus.api.DataBusBroker;
+import org.eclipse.lsp.cobol.domain.databus.api.DeadEventSubscriber;
+import org.eclipse.lsp.cobol.domain.databus.model.AnalysisFinishedEvent;
+import org.eclipse.lsp.cobol.domain.databus.model.RunAnalysisEvent;
+import org.eclipse.lsp.cobol.service.utils.CustomThreadPoolExecutor;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test {@link DefaultDataBusBroker}
+ */
+class DefaultDataBusBrokerTest {
+  @Test
+  void testTwoTypesOfEvents() {
+    AnalysisFinishedEventSubscriber analysisFinishedEventSubscriber = new AnalysisFinishedEventSubscriber();
+    RunAnalysisEventSubscriber runAnalysisEventSubscriber = new RunAnalysisEventSubscriber();
+    DeadSubscriber deadSubscriber = new DeadSubscriber();
+    DataBusBroker dataBus = new DefaultDataBusBroker(customThreadPoolExecutor, deadSubscriber);
+    dataBus.subscribe(analysisFinishedEventSubscriber);
+    dataBus.subscribe(runAnalysisEventSubscriber);
+    dataBus.postData(new AnalysisFinishedEvent());
+    dataBus.postData(new AnalysisFinishedEvent());
+    dataBus.postData(new RunAnalysisEvent());
+    assertEquals(2, analysisFinishedEventSubscriber.hits);
+    assertEquals(1, runAnalysisEventSubscriber.hits);
+    assertEquals(0, deadSubscriber.hits);
+  }
+
+  @Test
+  void testDeadSubscriber() {
+    DeadSubscriber deadSubscriber = new DeadSubscriber();
+    DataBusBroker dataBus = new DefaultDataBusBroker(customThreadPoolExecutor, deadSubscriber);
+    dataBus.postData(new RunAnalysisEvent());
+    assertEquals(1, deadSubscriber.hits);
+  }
+
+  @Test
+  void testUnsubscribe() {
+    RunAnalysisEventSubscriber runAnalysisEventSubscriber = new RunAnalysisEventSubscriber();
+    DeadSubscriber deadSubscriber = new DeadSubscriber();
+    DataBusBroker dataBus = new DefaultDataBusBroker(customThreadPoolExecutor, deadSubscriber);
+    dataBus.subscribe(runAnalysisEventSubscriber);
+    dataBus.postData(new RunAnalysisEvent());
+    assertEquals(1, runAnalysisEventSubscriber.hits);
+    assertEquals(0, deadSubscriber.hits);
+    dataBus.unSubscribe(runAnalysisEventSubscriber);
+    dataBus.postData(new RunAnalysisEvent());
+    assertEquals(1, runAnalysisEventSubscriber.hits);
+    assertEquals(1, deadSubscriber.hits);
+  }
+
+  // This executor runs everything in the same thread
+  private CustomThreadPoolExecutor customThreadPoolExecutor = new CustomThreadPoolExecutor() {
```

Comment:
Why not using already existing CustomThreadPoolExecutorService, just providing corePoolSize to 1, same for scheduled.

Reply:
Because CustomThreadPoolExecutorService creates one thread. And event handler will run in a separate thread, so I need to wait for it in the test and check the status.
But this implementation runs handler in the same thread as tests. And I don't need to sync it, everything is synchronous.